Could NOT find cuBLAS (missing CUBLAS_INCLUDE_DIR)

# Summary
Building with the `cublas` back-end fails to find `cuBLAS` library:
```
$ cmake -DBUILD_FUNCTIONAL_TESTS=ON -DENABLE_MKLGPU_BACKEND=OFF -DENABLE_MKLCPU_BACKEND=OFF -DENABLE_CUBLAS_BACKEND=ON -DENABLE_CURAND_BACKEND=OFF -DOPENCL_INCLUDE_DIR=/opt/khronos/ocl-headers/include -DREF_BLAS_ROOT=/opt/netlib/lapack/3.9.0 $SRCDIR
-- CMAKE_BUILD_TYPE: None, set to Release by default
-- The CXX compiler identification is Clang 12.0.0
-- Check for working CXX compiler: /opt/intel/llvm/2020.12.27-6ca33e2df283-cuda/bin/clang++
-- Check for working CXX compiler: /opt/intel/llvm/2020.12.27-6ca33e2df283-cuda/bin/clang++ -- works
<...>
-- Found CUDA: /opt/nvidia/cuda/10.2 (found suitable version "10.2", minimum required is "10.0") 
CMake Error at /usr/share/cmake-3.16/Modules/FindPackageHandleStandardArgs.cmake:146 (message):
  Could NOT find cuBLAS (missing: CUBLAS_INCLUDE_DIR)
Call Stack (most recent call first):
  /usr/share/cmake-3.16/Modules/FindPackageHandleStandardArgs.cmake:393 (_FPHSA_FAILURE_MESSAGE)
  cmake/FindcuBLAS.cmake:39 (find_package_handle_standard_args)
  src/blas/backends/cublas/CMakeLists.txt:22 (find_package)
<...>
```
causing the `cmake` configuration to fail. This is due to the fact that [`CMAKE_CUDA_TOOLKIT_INCLUDE_DIRECTORIES`](https://cmake.org/cmake/help/latest/variable/CMAKE_CUDA_TOOLKIT_INCLUDE_DIRECTORIES.html) is not defined by default.

# Version
0.1.0, a99dde887

# Environment
CUDA: cuda/10.2
Compiler: intel/llvm@6ca33e2 (compiled with CUDA support)

# Steps to reproduce
Setup environment and use the command in "Summary".

# Observed behavior
`cmake` configuration fails. (See "Summary".)

# Expected behavior
```
$ cmake <...> -DENABLE_CUBLAS_BACKEND=ON  <...>
<...>
-- Found cuBLAS: /opt/nvidia/cuda/10.2/targets/x86_64-linux/include
<...>
```

# Solution
The [preferred method](https://cmake.org/cmake/help/v3.15/module/FindCUDA.html?highlight=findcuda) is to add `enable_language(CUDA)` to [`cmake/FindcuBLAS.cmake`](https://github.com/oneapi-src/oneMKL/blob/master/cmake/FindcuBLAS.cmake):
```
diff --git a/cmake/FindcuBLAS.cmake b/cmake/FindcuBLAS.cmake
index 06fe6fe..f79d571 100644
--- a/cmake/FindcuBLAS.cmake
+++ b/cmake/FindcuBLAS.cmake
@@ -18,6 +18,7 @@
 #=========================================================================
 
 find_package(CUDA 10.0 REQUIRED)
+enable_language(CUDA)
 find_path(CUBLAS_INCLUDE_DIR "cublas_v2.h" HINTS ${CMAKE_CUDA_TOOLKIT_INCLUDE_DIRECTORIES})
 get_filename_component(SYCL_BINARY_DIR ${CMAKE_CXX_COMPILER} DIRECTORY)
 # the OpenCL include file from cuda is opencl 1.1 and it is not compatible with DPC++
```

--- a/cmake/FindcuBLAS.cmake
+++ b/cmake/FindcuBLAS.cmake
@@ -18,7 +18,6 @@
 #=========================================================================
 
 find_package(CUDA 10.0 REQUIRED)
-find_path(CUBLAS_INCLUDE_DIR "cublas_v2.h" HINTS ${CMAKE_CUDA_TOOLKIT_INCLUDE_DIRECTORIES})
 get_filename_component(SYCL_BINARY_DIR ${CMAKE_CXX_COMPILER} DIRECTORY)
 # the OpenCL include file from cuda is opencl 1.1 and it is not compatible with DPC++
 # the OpenCL include headers 1.2 onward is required. This is used to bypass NVIDIA OpenCL headers
@@ -27,8 +26,6 @@ HINTS
 ${OPENCL_INCLUDE_DIR}
 ${SYCL_BINARY_DIR}/../include/sycl/
 )
-find_library(CUBLAS_LIBRARY cublas)
-find_library(CUDA_DRIVER_LIBRARY cuda)
 # this is work around to avoid duplication half creation in both cuda and SYCL
 add_compile_definitions(CUDA_NO_HALF)
 
@@ -37,19 +34,18 @@ find_package(Threads REQUIRED)
 include(FindPackageHandleStandardArgs)
 find_package_handle_standard_args(cuBLAS
     REQUIRED_VARS
-        CUBLAS_INCLUDE_DIR
-        CUDA_INCLUDE_DIRS
-        CUBLAS_LIBRARY
+	CUDA_TOOLKIT_INCLUDE
+	CUDA_cublas_LIBRARY
         CUDA_LIBRARIES
-        CUDA_DRIVER_LIBRARY
+        CUDA_CUDA_LIBRARY
         OPENCL_INCLUDE_DIR
 )
 if(NOT TARGET ONEMKL::cuBLAS::cuBLAS)
   add_library(ONEMKL::cuBLAS::cuBLAS SHARED IMPORTED)
   set_target_properties(ONEMKL::cuBLAS::cuBLAS PROPERTIES
-      IMPORTED_LOCATION ${CUBLAS_LIBRARY}
-      INTERFACE_INCLUDE_DIRECTORIES "${OPENCL_INCLUDE_DIR};${CUDA_INCLUDE_DIRS}"
-      INTERFACE_LINK_LIBRARIES "Threads::Threads;${CUDA_DRIVER_LIBRARY};${CUDA_LIBRARIES}"
+      IMPORTED_LOCATION ${CUDA_cublas_LIBRARY}
+      INTERFACE_INCLUDE_DIRECTORIES "${OPENCL_INCLUDE_DIR};${CUDA_TOOLKIT_INCLUDE}"
+      INTERFACE_LINK_LIBRARIES "Threads::Threads;${CUDA_CUDA_LIBRARY};${CUDA_LIBRARIES}"
   )
 
 endif()

